File: seed/csharp-sdk/pagination/src/SeedPagination/Users/Requests/ListUsernamesRequest.cs
using SeedPagination.Core;

namespace SeedPagination;

public record ListUsernamesRequest
{
    /// <summary>
    /// The cursor used for pagination in order to fetch
    /// the next page of results.
    /// </summary>
    public string? StartingAfter { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-sdk/pagination/property-accessors/src/InlineUsers/InlineUsers/Requests/ListUsernamesRequest.php
<?php

namespace Seed\InlineUsers\InlineUsers\Requests;

use Seed\Core\Json\JsonSerializableType;

class ListUsernamesRequest extends JsonSerializableType
{
    /**
     * The cursor used for pagination in order to fetch
     * the next page of results.
     *
     * @var ?string $startingAfter
     */
    private ?string $startingAfter;

    /**
     * @param array{
     *   startingAfter?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    )
    {
        $this->startingAfter = $values['startingAfter'] ?? null;
    }

    /**
     * @return ?string
     */
    public function getStartingAfter(): ?string {
        return $this->startingAfter;}

    /**
     * @param ?string $value
     */
    public function setStartingAfter(?string $value = null): self {
        $this->startingAfter = $value;return $this;}
}
